test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -109,13 +121,39 @@ class test_ucp_sockaddr : public ucp_test {
         ep_params.sockaddr.addrlen = sizeof(*connect_addr);
         sender().connect(&receiver(), ep_params);
 
-        tag_send_recv(sender(), receiver());
-
-        /* wait for reverse ep to appear */
-        while (receiver().get_num_eps() == 0) {
+        if (wakeup) {
```

Comment:
there should not be any different path for "wait" in this function, only "wait()" function should be different.
why need to do wait/progress here if it's already done in wait()?

Reply:
the wait function takes care of progressing after the send-recv starts. waiting here is for the connection establishment part on the server side.

Reply:
why need to wait for connection establishment?